sig_finder/sig_finder.cc

```diff
@@ -0,0 +1,119 @@
+#include "absl/strings/ascii.h"
```

Comment:
small thing: what do you think about putting this in `main/lsp` where the other things like `NextMethodFinder` are?

Reply:
The code is copied from @froydnj PR https://github.com/sorbet/sorbet/pull/5169, and it's used outside of `lsp`. I haven't updated the Nathan's call site yet, but will do soon. That might be blocking moving it into the `lsp` package

Reply:
sounds good thanks, i didn't see that usage. i still think we probably don't want to make a new top-level folder for this. maybe something like `ast/sig_finder/`